R/test_wrapper.R

```diff
@@ -263,21 +263,30 @@ tidy.chisq_exploratory <- function(x, type = "observed") {
     ret <- ret %>% tibble::rownames_to_column(var = x$var1)
   }
   if (type == "residuals") {
+    obs_df <- as.data.frame(x$observed)
+    obs_df <- obs_df %>% tibble::rownames_to_column(var = x$var1)
+    obs_df <- obs_df %>% tidyr::gather(!!rlang::sym(x$var2), "observed", -!!rlang::sym(x$var1))
+
+    expected_df <- as.data.frame(x$expected)
+    expected_df <- expected_df %>% tibble::rownames_to_column(var = x$var1)
+    expected_df <- expected_df %>% tidyr::gather(!!rlang::sym(x$var2), "expected", -!!rlang::sym(x$var1))
+
     resid_df <- as.data.frame(x$residuals)
     resid_df <- resid_df %>% tibble::rownames_to_column(var = x$var1)
     resid_df <- resid_df %>% tidyr::gather(!!rlang::sym(x$var2), "residual", -!!rlang::sym(x$var1))
 
-    obs_df <- as.data.frame(x$observed)
-    obs_df <- obs_df %>% tibble::rownames_to_column(var = x$var1)
-    obs_df <- obs_df %>% tidyr::gather(!!rlang::sym(x$var2), "observed", -!!rlang::sym(x$var1))
+    resid_raw_df <- as.data.frame(x$observed - x$expected) # x$residual is standardized, but here, take raw difference between observed and expected. 
+    resid_raw_df <- resid_raw_df %>% tibble::rownames_to_column(var = x$var1)
+    resid_raw_df <- resid_raw_df %>% tidyr::gather(!!rlang::sym(x$var2), "residual_raw", -!!rlang::sym(x$var1))
 
-    raw_resid_df <- as.data.frame(x$observed - x$expected) # x$residual is standardized, but here, take raw difference between observed and expected. 
-    raw_resid_df <- raw_resid_df %>% tibble::rownames_to_column(var = x$var1)
-    raw_resid_df <- raw_resid_df %>% tidyr::gather(!!rlang::sym(x$var2), "raw_residual", -!!rlang::sym(x$var1))
+    resid_ratio_df <- as.data.frame((x$observed - x$expected)/x$expected) # x$residual is standardized, but here, take raw difference between observed and expected. 
+    resid_ratio_df <- resid_ratio_df %>% tibble::rownames_to_column(var = x$var1)
+    resid_ratio_df <- resid_ratio_df %>% tidyr::gather(!!rlang::sym(x$var2), "residual_ratio", -!!rlang::sym(x$var1))
 
-    ret <- obs_df %>% left_join(resid_df, by=c(x$var1, x$var2)) # join residual column 
-    ret <- ret %>% left_join(raw_resid_df, by=c(x$var1, x$var2)) # join raw_residual column
-    browser()
+    ret <- obs_df %>% left_join(expected_df, by=c(x$var1, x$var2)) # join expected column 
```

Comment:
There are 2 more left joins added. Do you have any performance measurement on this? I'm not sure how big the base data frame is going to be.

Reply:
This is after summarizing. So the data size would be at most hundreds. So I wouldn't worry about performance here.